packages/ifpack2/src/Ifpack2_Details_FastILU_Base_def.hpp

```diff
@@ -179,64 +181,65 @@ initialize()
     }
     Teuchos::TimeMonitor timeMonMetis (*timerMetis);
     #ifdef HAVE_IFPACK2_METIS
-    // reorder will convert both graph and perm/iperm to the internal METIS integer type
     idx_t nrows = localRowPtrsHost_.size() - 1;
-    metis_perm_  = MetisArrayHost(Kokkos::ViewAllocateWithoutInitializing("metis_perm"),  nrows);
-    metis_iperm_ = MetisArrayHost(Kokkos::ViewAllocateWithoutInitializing("metis_iperm"), nrows);
-
-    // copy ColInds to host
-    auto localColIndsHost_ = Kokkos::create_mirror(localColInds_);
-    Kokkos::deep_copy(localColIndsHost_, localColInds_);
-
-    // prepare for calling metis
-    idx_t nnz = localColIndsHost_.size();
-    MetisArrayHost metis_rowptr;
-    MetisArrayHost metis_colidx;
-
-    bool metis_symmetrize = true;
-    if (metis_symmetrize) {
-      // symmetrize
-      using OrdinalArrayMirror = typename OrdinalArray::host_mirror_type;
-      KokkosKernels::Impl::symmetrize_graph_symbolic_hashmap<
-        OrdinalArrayHost, OrdinalArrayMirror, MetisArrayHost, MetisArrayHost, Kokkos::HostSpace::execution_space>
-        (nrows, localRowPtrsHost_, localColIndsHost_, metis_rowptr, metis_colidx);
-
-      // remove diagonals
-      idx_t old_nnz = nnz = 0;
-      for (idx_t i = 0; i < nrows; i++) {
-        for (LocalOrdinal k = old_nnz; k < metis_rowptr(i+1); k++) {
-          if (metis_colidx(k) != i) {
-            metis_colidx(nnz) = metis_colidx(k);
-            nnz++;
+    if (nrows > 0) {
+      // reorder will convert both graph and perm/iperm to the internal METIS integer type
+      metis_perm_  = MetisArrayHost(Kokkos::ViewAllocateWithoutInitializing("metis_perm"),  nrows);
+      metis_iperm_ = MetisArrayHost(Kokkos::ViewAllocateWithoutInitializing("metis_iperm"), nrows);
+
+      // copy ColInds to host
+      auto localColIndsHost_ = Kokkos::create_mirror(localColInds_);
```

Comment:
```suggestion
      auto localColIndsHost_ = Kokkos::create_mirror_view(Kokkos::ViewAllocateWithoutInitializing("localColIndsHost_"), localColInds_);
```
create_mirror always allocates a new view, but create_mirror_view will only allocate if localColinds_ is not already host accessible.

Reply:
I'll go through the code, thank you @brian-kelley!!